app/src/main/java/com/sdp/movemeet/backend/providers/BackendInstanceProvider.java

```diff
@@ -0,0 +1,40 @@
+package com.sdp.movemeet.backend.providers;
+
+import androidx.annotation.VisibleForTesting;
+
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.storage.FirebaseStorage;
+
+public class BackendInstanceProvider {
+
+    @VisibleForTesting(otherwise=VisibleForTesting.PRIVATE)
+    public static final BackendInstanceProvider instance = new BackendInstanceProvider();
+
+    @VisibleForTesting(otherwise=VisibleForTesting.PRIVATE)
+    public static FirebaseFirestore firestore;
+    @VisibleForTesting(otherwise=VisibleForTesting.PRIVATE)
+    public static FirebaseStorage storage;
+    @VisibleForTesting(otherwise=VisibleForTesting.PRIVATE)
+    public static FirebaseDatabase database;
+
+    private BackendInstanceProvider() {
+        firestore = FirebaseFirestore.getInstance();
+        storage = FirebaseStorage.getInstance();
+        database = FirebaseDatabase.getInstance();
+    }
+
+
```

Comment:
Couldn't the BackendInstanceProvider also provide a getter for the authentication, instead of defining a whole class for the authentication ?

Reply:
It is again an import due to a merge with kepler 😅